osgar/tools/log2map.py

```diff
@@ -20,16 +20,28 @@ def main():
     parser.add_argument('logfile', help='recorded log file')
     parser.add_argument('--out', help='map output file')
     parser.add_argument('--pose2d', help='stream ID for pose2d messages', default='app.pose2d')
+    parser.add_argument('--camera', help='stream ID with image data')
     args = parser.parse_args()
 
     out_filename = args.out
     if out_filename is None:
-        out_filename = os.path.join(os.path.dirname(args.logfile), 'world.jpg')
+        out_filename = os.path.join(os.path.dirname(args.logfile),
+                                    os.path.basename(args.logfile).split('.')[0] + '.jpg')
 
     pts = []
-    for dt, channel, data in LogReader(args.logfile, only_stream_id=lookup_stream_id(args.logfile, args.pose2d)):
-        pose = deserialize(data)
-        pts.append(pose[:2])
+    pose_id = lookup_stream_id(args.logfile, args.pose2d)
+    streams = [pose_id]
+    if args.camera is not None:
+        streams.append(lookup_stream_id(args.logfile, args.camera))
+    valid_motion = True
+    prev = None
+    for dt, channel, data in LogReader(args.logfile, only_stream_id=streams):
+        if channel == pose_id and valid_motion:
+            pose = deserialize(data)
+            pts.append(pose[:2])
+        else:
+            valid_motion = prev != data
+            prev = data
```

Comment:
What exactly is this `valid_motion` stuff?

Reply:
Maybe `valid_odometry` would be more suitable? Basically if the video from camera is still, but robot odometry is moving (typically out of world bounds) then something is wrong and you do not want to draw this very long line. On the other hand, as discussed on the last call, it depends on the environment and it looks like it behaves differently locally and on AWS ... also maybe the robot is on AWS slightly moving ...

Reply:
The thing is that I was not sure what the intention of the code was. The comment solves that, thanks.